x/upgrade/abci.go

```diff
@@ -22,16 +22,21 @@ import (
 // skipUpgradeHeightArray is a set of block heights for which the upgrade must be skipped
 func BeginBlocker(k keeper.Keeper, ctx sdk.Context, _ abci.RequestBeginBlock) {
 	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyBeginBlocker)
+
 	plan, found := k.GetUpgradePlan(ctx)
 	if !found {
+		verifyAppliedHandlerPresent(ctx, k)
```

Comment:
I think we can call it at the very beginning and only once:
```
if !downgradeVerified {
    downgradeVerified = true // this can be global in the module
    lastAppliedPlan, _ := k.GetNextOrLastUpgrade(ctx)
    if lastAppliedPlan != "" && !k.HasHandler(lastAppliedPlan) {
        panic(fmt.Sprintf("Wrong app version, upgrade handler for %s upgrade plan was already applied", lastAppliedPlan))
    }
}
```

Note:
+ we don't need the `verifyAppliedHandlerPresent` function - I think putting it the code here will simplify reading.